plugins/action/prettytable.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, quidame <quidame@poivron.org>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional
+
+__metaclass__ = type  # pylint: disable=C0103
+
+from ansible.errors import AnsibleError, AnsibleUndefinedVariable
+from ansible.module_utils._text import to_text
+from ansible.plugins.action import ActionBase
+
+try:
+    import prettytable
+
+    HAS_PRETTYTABLE = True
+except ImportError:
+    HAS_PRETTYTABLE = False
+
+
+class ActionModule(ActionBase):
+    """Print prettytable from list of dicts."""
+
+    TRANSFERS_FILES = False
+    _VALID_ARGS = frozenset(
+        ("data", "column_order", "header_names", "column_alignments")
+    )
+    _VALID_ALIGNMENTS = {"left", "center", "right", "l", "c", "r"}
+
+    def run(
+        self, tmp: Optional[str] = None, task_vars: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, Any]:
+        if task_vars is None:
+            task_vars = dict()
+
+        result = super(ActionModule, self).run(tmp, task_vars)  # pylint: disable=R1725
+        del tmp  # tmp no longer has any effect
+
+        try:
+            self._handle_options()
+
+            if not HAS_PRETTYTABLE:
+                raise AnsibleError(
+                    'You need to install "prettytable" python module before running this module'
+                )
+
+            data = self._task.args.get("data")
+            if data is None:  # Only check for None, allow empty list
+                raise AnsibleError("The 'data' parameter is required")
+
+            if not isinstance(data, list):
+                raise AnsibleError(
+                    "Expected a list of dictionaries, got a string"
+                    if isinstance(data, (str, bytes))
+                    else f"Expected a list of dictionaries, got {type(data).__name__}"
+                )
+
+            if data and not all(isinstance(item, dict) for item in data):
+                invalid_item = next(item for item in data if not isinstance(item, dict))
+                raise AnsibleError(
+                    "All items in the list must be dictionaries, got a string"
+                    if isinstance(invalid_item, (str, bytes))
+                    else f"All items in the list must be dictionaries, got {type(invalid_item).__name__}"
+                )
```

Comment:
This is not wrong, but personally I think it is over-engineered. 

I think something like:
```suggestion
            if not isinstance(data, list):
                raise AnsibleError("Parameter 'data' must be a list")

            if data and not all(isinstance(item, dict) for item in data):
                raise AnsibleError("All elements in 'data' must be dictionaries")
```
would get the job done and it would make the code simpler to read and maintain.